datamodel.customer-survey.xml

```diff
@@ -1922,6 +1925,59 @@ EOF
 				throw new Exception(Dict::S('Survey-email-notsent').': '.implode(', ', $aErrors));
 		}
 	}]]></code>
+        </method>
+        <method id="ApplyParamsToContent">
+          <comment>/**
+	 * @param string $sContent
+	 * @param Contact $oContact
+	 * @param ?string $sQuizUrl
+	 *
+	 * @return string Content with placeholders replaced.
+	 */</comment>
+          <static>false</static>
+          <access>protected</access>
+          <type>Custom</type>
+          <code><![CDATA[	protected function ApplyParamsToContent($sContent, $oContact, $sQuizUrl)
+{
+  $aParams = [];
+  $aParams['recipient->object()'] = $oContact;
+  if (false === is_null($sQuizUrl)){
+    $aParams['quiz_url'] = $sQuizUrl;
+  }
+  $aParams['this->object()'] = $this;
+
+  $sContent = MetaModel::ApplyParams($sContent, $aParams);
+
+  $this->AppendLinkToContent($sContent, $sQuizUrl);
```

Comment:
I think the ordering of these calls needs to be reversed?

Reply:
Was a enormous bug, I wrote very buggy code, my apologies.
Fixed in d31a92b2